independent-projects/resteasy-reactive/server/vertx/src/main/java/org/jboss/resteasy/reactive/server/vertx/AppendBuffer.java

```diff
@@ -1,53 +1,81 @@
 package org.jboss.resteasy.reactive.server.vertx;
 
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
 import java.util.ArrayDeque;
 import java.util.Objects;
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.ByteBufAllocator;
 import io.netty.buffer.CompositeByteBuf;
+import io.netty.buffer.PooledByteBufAllocator;
+import io.netty.util.internal.PlatformDependent;
+import io.vertx.core.buffer.impl.PartialPooledByteBufAllocator;
+import io.vertx.core.buffer.impl.VertxByteBufAllocator;
 
 /**
  * It is a bounded (direct) buffer container that can keep on accepting data till {@link #capacity} is exhausted.<br>
  * In order to keep appending on it, it can {@link #clear} and consolidate its content as a {@link ByteBuf}.
  */
 final class AppendBuffer {
-    private final ByteBufAllocator allocator;
+
+    private static final MethodHandle virtualMh = PlatformDependent.javaVersion() >= 21 ? findVirtualMH() : null;
+
+    private static MethodHandle findVirtualMH() {
+        try {
+            return MethodHandles.publicLookup().findVirtual(Thread.class, "isVirtual",
+                    MethodType.methodType(boolean.class));
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    private static boolean isVirtualThread() {
+        if (virtualMh == null) {
+            return false;
+        }
+        try {
+            return (boolean) virtualMh.invokeExact(Thread.currentThread());
```

Comment:
@galderz @zakkak can we verify that this one is as fast as a direct call in native-image?
I want to make sure we don't pay for this given that is in the fastest-path

Reply:
@franz1981 I'll look into it.